src/lib/orion_websocket/parser.cpp

```diff
@@ -160,7 +160,7 @@ const char *ws_parser_message
    * and add status code (3) and the final null character
    */
   char *json = (char *) malloc(strlen(tmpl) - 6 + strlen(headers) + msg.size() + 3 + 1);
-  sprintf(json, tmpl, headers, msg.c_str(), statusCode);
+  sprintf(json, tmpl, headers, msg.empty() ? "\"\"" : msg.c_str(), statusCode);
```

Comment:
Yes, you're right, the standard C++98 (21.3.6 basic_string string operations) define that, but in the C++11 standard (21.4.7.1 basic_string accessors) does not show any restriction about it. Now if you want use C++11 in the future (which would be great), this is may be is the better solution. But I do not complete sure.